src/Olm.cpp

```diff
@@ -70,6 +76,22 @@ handle_to_device_messages(const std::vector<mtx::events::collections::DeviceEven
                                   e.what(),
                                   j_msg.dump(2));
                         }
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationAccept)) {
+                        ChatPage::instance()->recievedDeviceVerificationAccept(msg);
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationRequest)) {
+                        ChatPage::instance()->recievedDeviceVerificationRequest(msg);
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationCancel)) {
+                        ChatPage::instance()->recievedDeviceVerificationCancel(msg);
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationKey)) {
+                        ChatPage::instance()->recievedDeviceVerificationKey(msg);
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationMac)) {
+                        ChatPage::instance()->recievedDeviceVerificationMac(msg);
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationStart)) {
+                        ChatPage::instance()->recievedDeviceVerificationStart(msg);
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationReady)) {
+                        ChatPage::instance()->recievedDeviceVerificationReady(msg);
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationDone)) {
+                        ChatPage::instance()->recievedDeviceVerificationDone(msg);
```

Comment:
I think you need to find at least a temporary solution to the done messages not being sent by some clients:

![grafik](https://user-images.githubusercontent.com/16312324/85564871-27217680-b61e-11ea-942e-2a1937de7cd0.png)

Reply:
Yeah, just pushed a commit fixing that

Reply:
Your commit does fix, that the verification doesn't spin forever. It however also cancels the verification on the other side :D

Reply:
It seems like your fix actually makes the verification succeed already, when the other party accepts, without me having to accept. That is probably bad :3